### PR TITLE
Exception is now thrown when creating wrong diagram type in SysML

### PR DESCRIPTION
**Pull request by FranciscoTGouveia**
### PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/main/CONTRIBUTING.md)
- [x] I have read, and I understand the GNOME [Code of Conduct](https://wiki.gnome.org/Foundation/CodeOfConduct)

### PR Type
What kind of change does this PR introduce?

- [x] Bug fix
- [ ] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: #3194

### What is the new behavior?

Now, when creating a wrong diagram type in SysML a notification pops.

Screenshot from 2024-03-25 17-50-22

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### Other information

### CODE REVIEW

File: gaphor/SysML/diagramtype.py [hunk 1]
```diff
@@ -29,13 +29,16 @@ def allowed(self, element) -> bool:
...
+            try:
+                d = next(d for d in self._defaults if isinstance(element, d.from_type))
+                assert d.to_type in self._allowed_types
+                new_element = element_factory.create(d.to_type)
+                new_element.name = d.name
+                if element:
+                    change_owner(element, new_element)
+                element = new_element
+            except:
+                raise TypeError(f"Cannot create \"{self.name}\" in SysML profile")
```
[the remark this answers is not in the document]
Comment (FranciscoTGouveia) [reply gaphor/SysML/diagramtype.py]: Done!

**Comment by amolenaar** [comment]
Looks good. You may want to run `pre-commit run -a` on the code and fix a few linting errors.

Note to self: we should probably mention this in the developer docs.

**Review by amolenaar — APPROVED** [review]

**Comment by amolenaar** [comment]
@all-contributors add @FranciscoTGouveia for code.

**Comment by amolenaar** [comment]
@FranciscoTGouveia Thanks for the fix. Do you have any feedback on how easy/hard it was do create your first PR? Can we make the process more smooth?

**Comment by FranciscoTGouveia** [comment]
Hi! Thanks for accepting the PR. Overall I think it was a smooth experience, and look forward to do more contributions. I agree with adding `pre-commit` do the developer docs. I've thought that running `black` was enough.

**Comment by amolenaar** [comment]
@FranciscoTGouveia Thanks! I created #3236 to clarify the contribution process. Does it cover it enough?

**Comment by FranciscoTGouveia** [comment]
Yes, it looks great!

**Pull request merged**